src/types.md

```diff
@@ -84,6 +86,13 @@ unsigned bytes holding a sequence of UTF-8 code points. Since `str` is a
 [dynamically sized type], it is not a _first-class_ type, but can only be
 instantiated through a pointer type, such as `&str`.
 
+## Never type
+
+The never type `!` is a type with no values, representing the result of
+computations that never complete. Although it has size `0` the empty bit
+pattern is not a valid representation. Expressions of type `!` can be coerced
```

Comment:
I'm not sure what the comment about a representation is intended to mean; since it has size 0, it has no representation, I thought?

Reply:
An n-bit type has 2^n possible bit-patterns that can represent values. `bool` is represented as a byte, but only two of the 256 different bit-patterns are valid values: `[00000000]` (false) and `[00000001]` (true).

`()` is size 0 which means it has one bit-pattern: the empty array of bits `[]`. In the case of `()` this represents the value `()`.

`!` is size 0 but it the empty array of bits does not represent any value.

Reply:
Ah, right, I see what you're saying. Is that actually true, though? It seems to me that it's more accurate to say that `()` has one value, but no representations. Rust generally doesn't try to encode a representation of `()`, for instance, `Vec<()>` is effectively just a length field.

I'd probably just drop the sentence; I don't think it adds more clarity.

Reply:
Okay, fixed.